core/src/avm1/globals/array.rs

```diff
@@ -674,70 +674,61 @@ fn qsort<'gc>(
         return Ok(());
     }
 
-    // Fast-path for 2 elements.
-    if let [(_, a), (_, b)] = &elements {
-        if compare_fn(activation, a, b, options)?.is_gt() {
-            elements.swap(0, 1);
+    // Stack for storing subarray boundaries (start and end).
+    let mut stack: Vec<(usize, usize)> = Vec::new();
+
+    stack.push((0, elements.len() - 1));
+
+    while let Some((low, high)) = stack.pop() {
+        if low >= high {
```

Comment:
`>` isn't normally possible, right? Can we `assert` that?

Reply:
> `>` isn't normally possible, right?

Yes, you're right! Should I use `assert!` or `debug_assert!` for this?

Reply:
`assert` does sound nice, unless it kills the performance back.

Reply:
I think we can't replace the normal check with assert because low >= high can occur when the pivot partition results in a subarray with no elements. (e.g., [5, 5, 5, 5, 5])

Reply:
I know `low == high` can occur, but I meant that specifically `low > high`  shouldn't ever occur.